halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/PersistentStorageController.java

```diff
@@ -82,6 +83,49 @@
     builder.setRevert(() -> halconfigParser.undoChanges());
     builder.setSave(() -> halconfigParser.saveConfig());
 
-    return DaemonTaskHandler.submitTask(builder::build, "Edit storage settings");
+    return DaemonTaskHandler.submitTask(builder::build, "Set current persistent store");
+  }
+
+  @RequestMapping(value = "/persistentStore/{persistentStoreName:.+}", method = RequestMethod.GET)
+  DaemonTask<Halconfig, PersistentStore> getPersistentStore(@PathVariable String deploymentName,
```

Comment:
Easiest way to ensure `enable/disable` only shows up in the generated config is probably to cast the whole storage object into a `Map` using the `@Autowired ObjectMapper objectMapper`, and then write `.put("enabled", val)` for the necessary providers. Or you could subclass that storage object and then copy the fields over, with `enabled` being the only addition to that subclass.